conftest.py

```diff
@@ -544,54 +544,56 @@ def cassandra_dir_and_version(config):
+    if has_mark_fn(item, mark):
+        if skip_marked:
             logger.info("SKIP: Skipping %s because it is marked with %s" % (item, mark))
             return True
-    else:
-        if include_other:
-            return False
         else:
+            return False
+    else:
+        if skip_non_marked:
             logger.info("SKIP: Skipping %s because it is not marked with %s" % (item, mark))
             return True
+        else:
+            return False
 
+def _can_run_resource_intensive_tests(dtest_config, sufficient_resources):
+    return (dtest_config.force_execution_of_resource_intensive_tests
+            or dtest_config.only_resource_intensive_tests
+            or sufficient_resources)
 
-def is_skippable(item,
-                 include_upgrade_tests,
-                 include_non_upgrade_tests,
-                 include_resource_intensive_tests,
-                 include_non_resource_intensive_tests,
-                 include_vnodes_tests,
-                 include_no_vnodes_tests,
-                 include_no_offheap_memtables_tests):
-
-    skippable = False
+def is_skippable(item, dtest_config, sufficient_resources):
+    can_run_resource_intensive_tests = _can_run_resource_intensive_tests(dtest_config, sufficient_resources)
 
-    skippable = skippable or _is_skippable(item, "upgrade_test", include_upgrade_tests, include_non_upgrade_tests)
-    skippable = skippable or _is_skippable(item, "resource_intensive", include_resource_intensive_tests, include_non_resource_intensive_tests)
-    skippable = skippable or _is_skippable(item, "vnodes", include_vnodes_tests, True)
-    skippable = skippable or _is_skippable(item, "no_vnodes", include_no_vnodes_tests, True)
-    skippable = skippable or _is_skippable(item, "no_offheap_memtables", include_no_offheap_memtables_tests, True)
-    skippable = skippable or _is_skippable(item, "depends_driver", False, True)
+    skip_upgrade_tests = not dtest_config.execute_upgrade_tests and not dtest_config.execute_upgrade_tests_only
+    skip_non_upgrade_tests = dtest_config.execute_upgrade_tests_only
+    skip_resource_intensive_tests = not can_run_resource_intensive_tests or dtest_config.skip_resource_intensive_tests
+    skip_non_resource_intensive_tests = dtest_config.only_resource_intensive_tests
+    skip_vnodes_tests = not dtest_config.use_vnodes
+    skip_no_vnodes_tests = dtest_config.use_vnodes
+    skip_no_offheap_memtables_tests = dtest_config.use_off_heap_memtables
 
-    return skippable
+    return (_is_skippable(lambda i, m: has_mark(i, m) or any_class_has_mark(i, m), item, "upgrade_test",
+                          skip_upgrade_tests, skip_non_upgrade_tests)
+            or _is_skippable(has_mark, item, "resource_intensive",
+                             skip_resource_intensive_tests, skip_non_resource_intensive_tests)
+            or _is_skippable(has_mark, item, "vnodes", skip_vnodes_tests, skip_non_marked=False)
```

Comment:
this is not obvious for me...
we are changing:
`_is_skippable(item, "vnodes", include_vnodes_tests)` to
`_is_skippable(has_mark, item, "vnodes", skip_vnodes_tests, skip_non_marked=False)`
so include -> skip.
maybe I need to take a look again.

Reply:
No, it is changing:
`_is_skippable(item, "vnodes", include_vnodes_tests, True)`
I.e., it adds the has mark functions and flips the last two arguments: `include_vnodes_tests`-> `skip_vnodes_tests` and `True` -> `skip_non_marked=False`.
I find the original naming: `skippable(..., include)`, confusing. It is not intuitive to me what `include` means in the context of `skip` without understanding the implementation.